tracing-subscriber/src/registry/extensions.rs

```diff
@@ -0,0 +1,182 @@
+// taken from https://github.com/hyperium/http/blob/master/src/extensions.rs.
```

Comment:
I'd love a comment here explaining what on earth extensions are, and why we need them.

Reply:
n.b. that it probably shouldn't be in the module-level docs, since they aren't public